### PR TITLE
✨ 210 - add collaborator apps search endpoint

### PR DESCRIPTION
**Pull request by anncatton**
Adds endpoint to search a logged-in user's applications where they are a collaborator.
Small fix for typescript error on catch clause.

### CODE REVIEW

File: src/routes/applications.ts [hunk 1]
```diff
@@ -334,6 +339,16 @@ const createApplicationsRouter = (
     }),
   );
 
+  router.get(
+    '/collaborators/applications',
```
Comment (anncatton) [inline src/routes/applications.ts:343]: not sure about the best way to name this endpoint?

File: src/domain/service.ts [hunk 2]
```diff
@@ -458,6 +458,24 @@ export async function search(params: SearchParams, identity: Identity): Promise<
...
+  // using ego token email matched against collaborator googleEmail
+  const apps = await ApplicationModel.find({
+    state: 'APPROVED',
+    'sections.collaborators.list.info.googleEmail': identity.tokenInfo.context.user.email,
+  });
+
+  return apps.map(
+    (app: ApplicationDocument) =>
+      ({
+        appId: `${app.appId}`,
```
Comment (ciaranschutte) [inline src/domain/service.ts:472]: @anncatton 
do we need `institution` also? 
https://app.zeplin.io/project/5a69fad2bde6330a929157fd/screen/60ca5a69e82750bf39b200d1
I see there are two different institutions

File: src/routes/applications.ts (hunk 1 above)
Comment (blabadi) [reply src/routes/applications.ts:343]: I think it's fine for now

Comment (anncatton) [reply src/routes/applications.ts:343]: oops yes I forgot. will add now 👍

Comment (anncatton) [reply src/routes/applications.ts:343]: @blabadi added route to swagger

File: src/resources/swagger.yaml [hunk 3]
```diff
@@ -381,14 +379,43 @@ paths:
               schema:
                 $ref: '#/components/schemas/Collaborator'
 
+  /collaborators/applications/:
```
Comment (anncatton) [inline src/resources/swagger.yaml:382]: route added here, everything else is formatting

**Review by blabadi — APPROVED** [review]

**Pull request merged**